src/app/static/src/app/components/HintTreeSelect.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+    <treeselect
+    :key="reRender"
+    :options="options"
+    :model-value="modelValue"
+    :multiple="multiple"
+    @update:model-value="input">
+    <!-- way to pass down all slots -->
+    <template v-for="(_, slot) in $slots" v-slot:[slot]="scope">
+        <slot :name="slot" v-bind="scope || {}" />
+    </template>
+    </treeselect>
+</template>
+<script lang="ts">
+    import { PropType, defineComponent } from "vue";
+    import Treeselect from "vue3-treeselect";
+
+    // This wrapper was necessary as this third party treeselect was not
+    // updating its value when an external update was being made, e.g.
+    // when dataSource changed in Choropleth tab, the filters' values
+    // would not update.
+    
+    // Here we force this treeselect component to re-render if an external
+    // change has been made. If an internal change is made (like user
+    // selecting a value) then the lastEmittedValue keeps track of it so
+    // we can prevent re-render.
+
+    export default defineComponent({
+        components: {
+            Treeselect
+        },
+        props: {
+            options: {
+                type: Array,
+                required: true
+            },
+            modelValue: {
+                type: Array as PropType<string | string[] | null | undefined>,
+                required: true
+            },
+            multiple: {
+                type: Boolean
+            }
+        },
+        data() {
+            return {
+                // this is an integer that toggles between 1 and 0 to cause
+                // key to change and force re-render
+                reRender: 0,
+                lastEmittedValue: "" as string | string[] | null | undefined
+            }
+        },
+        methods: {
+            input(value: string[]) {
+                this.lastEmittedValue = value;
+                this.$emit("update:model-value", value)
+            }
+        },
+        watch: {
+            modelValue: {
+                handler: function(newVal, oldVal) {
+                    // Multi-select component already seems to work fine so we want to
+                    // not re-render those. We only want to re-render if:
+                    // newVal === null : to replace value with "Not used" in treeselect
+                    // newValStringCase : to re-render when we pass string into treeselect
+                    // newValArrayCase : to re-render when we pass array of length 1 into treeselect
+
+                    const newValStringCase = (typeof newVal === "string") && this.lastEmittedValue !== newVal;
+                    const newValArrayCase = newVal instanceof Array && newVal.length === 1 && this.lastEmittedValue !== newVal[0];
```

Comment:
I don't quite see how this works together with the parameter type of `input` being a string[] - if last emitted was ["someValue"] and ["someValue"] is the newVal here, `newValArrayCase` would wrongly be true here since ["someValue"] !== "someValue". Should the `value` parameter type of `input` actually be `string` - or does the interface specify `string[]` but you actually get strings? Should `string` also be the type of `lastEmittedValue`?

Reply:
Sorry I should have really included an explanation for this in the PR comment. After some investigation, I found the third-party treeselect component *always* emits a `string` on input. When things happen in our store though, some `newVal`s are `string[]` of length 1 and some are `string`. Not sure where this inconsistency came from but equally I would say this is not worth fixing as:
1. There might actually be a reason for this
2. This would involve checking every state that stores a possible treeselect value and potentially altering the type, which seems OTT for a render issue 😆 
So I have included these two cases!